modules/travel_pay/app/services/travel_pay/appointments_service.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module TravelPay
+  class AppointmentsService
+    ##
+    # gets all appointments and finds the singular BTSSS appointment that matches the provided datetime
+    # @params: datetime string ('2024-01-01T12:45:34.465Z')
+    #
+    # @return {
+    # 'id' => 'string',
+    # 'appointmentSource' => 'string',
+    # 'appointmentDateTime' => 'string',
+    # 'appointmentName' => 'string',
+    # 'appointmentType' => 'string',
+    # 'facilityName' => 'string',
+    # 'serviceConnectedDisability' => int,
+    # 'currentStatus' => 'string',
+    # 'appointmentStatus' => 'string',
+    # 'externalAppointmentId' => 'string',
+    # 'associatedClaimId' => string,
+    # 'associatedClaimNumber' => string,
+    # 'isCompleted' => boolean,
+    # }
+    #
+    #
+    def get_appointment_by_date_time(veis_token, btsss_token, params = {})
+      faraday_response = client.get_all_appointments(veis_token, btsss_token, { 'excludeWithClaims' => true })
+      raw_appointments = faraday_response.body['data'].deep_dup
+
+      appointment = find_by_date_time(params['appt_datetime'], raw_appointments)
+
+      {
+        data: appointment
+      }
+    end
+
+    private
+
+    def find_by_date_time(date_string, appointments)
+      if date_string.present?
+
+        appointments.find do |appt|
+          date_string == appt['appointmentDateTime']
```

Comment:
You mentioned in Slack that we're not parsing or concerned about `Date:Error`s anymore because 
> the date being passed in is going to (or it should anyway) match whatever appointment date is in BTSSS, since they both pull from the same system

Yes the source data is a common system, but there are more systems involved, each with their own rules and serializers. Here's a quick picture:
```mermaid
flowchart LR
  VistA --> MAS
  MAS --> vets-api_Appointments
  VistA --> BTSSS_Plugins
  BTSSS_Plugins --> Travel_Pay_API
  vets-api_Appointments --> vets-api_Travel_Pay
  Travel_Pay_API --> vets-api_Travel_Pay
```

Because of all that is involved between the appointments source of truth and the travel pay code, I think we should go back to parsing the date. And if we parse the date, I have thoughts on why we should throw an ArgumentError.

Reply:
To clarify, if any of those intermediary systems changes the date format, we'd want to turn in into a `DateTime` object by parsing it so we're comparing the actual date, not  the strings themselves.

Reply:
Yeah, that's a fair point. Thanks for clarifying.